### PR TITLE
[serve] Add `log_to_stderr` option to logger and improve internal logging

### LINKED ISSUE #33203: [serve] Add controller logs for when replicas finish starting up
Currently trying to help a user debug an issue and we have logs for when replicas are started but not when they finish initializing. This is important context.

For example we have:
```
 155 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,180 controller 979 deployment_state.py:625 - Actor for replica D#DYFbUC crashed, marking it unhealthy immediately.
 154 2023-03-10T11:14:57-06:00 WARNING 2023-03-10 09:14:57,187 controller 979 deployment_state.py:1506 - Replica D#DYFbUC of deployment D failed health check, stopping it.
 153 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,327 controller 979 deployment_state.py:1310 - Adding 1 replica to deployment 'D'.
```

Would be good to also have:
```
 153 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,327 controller 979 deployment_state.py:1310 - Starting replica <REPLICA_ID> for deployment 'D'.
 153 2023-03-10T11:14:57-06:00 INFO 2023-03-10 09:14:57,327 controller 979 deployment_state.py:1310 - Replica <REPLICA_ID> for deployment 'D' finished initializing after <TIME_S>.
```

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

A few improvements to logging:
- add the ability to log certain messages only to log files and not to `stderr` using `extra={"log_to_stderr": False}`. This should be used going forward for things we don't want to spam the console but do want to show up in cluster logs.
- add some more log lines in relevant places (HTTP proxies getting updates, deploy task being launch and finishing, more logs in controller).
- remove the deprecated `"ray"` logger logic & related test case.

Closes https://github.com/ray-project/ray/issues/33203

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ollie-iterators** [comment]
The error with actor was caused by a recent pull request. See https://flakey-tests.ray.io/.

File: python/ray/serve/_private/http_proxy.py [hunk 1]
```diff
@@ -401,7 +404,8 @@ async def __call__(self, scope, receive, send):
                 method=scope["method"],
                 status=str(status_code),
                 latency_ms=latency_ms,
-            )
+            ),
+            extra={"log_to_stderr": False}
```
Comment (zcin) [inline python/ray/serve/_private/http_proxy.py:408]: Should this be debug?

**Review by zcin — COMMENTED** [review]
Looks great! I noticed a lot of new logs added/bumped that I would appreciate for debugging as well.

**Comment by ollie-iterators** [comment]
The error with the code is causing many tests to not be run when pull requests are merged

**Review by sihanwang41 — APPROVED** [review]

File: dashboard/modules/serve/serve_agent.py [hunk 2]
```diff
@@ -62,6 +62,7 @@ async def get_all_deployments(self, req: Request) -> Response:
                 # Task failure sometimes are due to GCS
                 # failure. When GCS failed, we expect a longer time
                 # to recover.
+                logger.exception("Failed to fetch config from the controller.")
```
[the remark this answers is not in the document]
Comment (edoakes) [reply dashboard/modules/serve/serve_agent.py:65]: Good point, they should probably all be exceptions. Let me update.

[the remark this answers is not in the document]
Comment (edoakes) [reply dashboard/modules/serve/serve_agent.py:65]: But to be clear in general the idea is that things that are user-initiated (such as a bad config) should be an info/warning level log. Errors/exceptions should be reserved for system errors (such as the controller being dead).

Also, `logger.exception` will print a nice stacktrace in addition to the error.

**Pull request merged**